File: задание 1/main.py
N = int(input("Input length: "))
M = int(input("Input width: "))

commands = input("Input all commands: ")

x = y = 0

def fun(commands):
    global x,y

    for command in commands:
        if command == 'L':
            x -= 1
        elif command == 'R':
            x += 1
        elif command == 'D':
            y += 1
        elif command == 'U':
            y -= 1
        else:
            raise Exception("Invalid command - {}".format(command))

        if abs(y) >= N or abs(x) >= M:
            print("No")
            return False

    return True

if fun(commands):
    print("({},{})".format(M - abs(x), N - abs(y)))

File: задание 1/main.swift
import Foundation

print("Input length: ", terminator: "")
let N = Int(readLine()!)!

print("Input width: ", terminator: "")
let M = Int(readLine()!)!

print("Input all commands: ", terminator: "")
let commands = readLine()!

var x = 0
var y = 0

for command in commands {
    switch command {
    case "L":
        x -= 1
    case "R":
        x += 1
    case "D":
        y += 1
    case "U":
        y -= 1
    default:
        fatalError("Invalid command - \(command)")
    }

    if abs(y) >= N || abs(x) >= M {
        print("No")
        exit(1)
    }
}

print("(\(M - abs(x)), \(N - abs(y)))")
